### ISSUE TITLE
Allow localization of UI

### ISSUE THREAD

**Issue by yarons** [apache/airflow#9864]
**Description**

I want to be able to translate the UI and possibly also adapt it to RTL (for Hebrew, Arabic, Persian, etc.)

**Use case / motivation**

I think that Airflow is very friendly in terms of UI and allowing the users to use Airflow in their own language might increase adoption and popularity and may also increase using this UI as data pipeline management tool for non developers.

**Related Issues**

I think this process should be fairly easy since the code is well designed and written.
The goto approach would probably be `gettext` even though it's pretty ancient and clumsy it's still very popular.
Other solutions might include `Python-i18next`, `Python Babel` etc. which I'm not very familiar with.
There are numerous advantages to the newer frameworks in terms of runtime efficiency and making it easy for translators to do their task once (instead of repeating strings like "Cancel", "OK", etc.).

Once the i18n process is done it's possible to use tools like [Weblate](https://weblate.org) to make sure the translators won't have to deal with code, git, etc.

I'm hoping localization will improve this tool in many ways and will bring new possibilities.

**Comment by mik-laj** [comment]
We are open to supporting multiple languages. There have been attempts to do this before, but I have not seen the final results of this work. Could you please prepare a POC and translate 1/2 views?  We are open to supporting multiple languages. There have been attempts to do this before, but I have not seen the final results of this work. Could you please prepare a POC and translate 1/2 views?

**Comment by JeffryMAC** [comment]
@yarons  one of the attempt to do this https://github.com/apache/airflow/pull/5344

**Comment by bbovenzi** [comment]
This is certainly a feature we're looking into for the new UI in AIP-38 since it's easier to include i18n at the start as opposed to adding it later on.

**Comment by Raydrrrr1234** [comment]
I would agree with adding a language parameter to the airflow.cfg file that indicates the webserver UI uses different language.

**Comment by XuanmoFeng** [comment]
i  will push the translate branch next week <img width="1759" alt="image" src="https://user-images.githubusercontent.com/17818323/235900695-3d8abaf3-5e5a-47be-8fbd-d22a4d08cdd5.png">

**Comment by mahdi-mirz** [comment]
I wanna translate UI of airflow to Persian, anyone have suggestion ?

**Comment by bbovenzi** [comment]
With the new UI in Airflow 3.0, this should be more possible to do. Definitely a good candidate for a 3.1+ feature

**Comment by RoyLee1224** [comment]
Hi @bbovenzi 

I've prepared an initial proof-of-concept and opened a Pull Request #50626

This PR introduces `react-i18next` and applies translations to static text in the Dashboard and Navigation sections. I would greatly appreciate your feedback on the general approach before I move on to more complex aspects (like pluralization and dynamic content, detailed in the PR).

Thank you!

**Comment by RoyLee1224** [comment]
Hi, @bbovenzi 

I'd like to help on this issue. Could you assign it to me?

**Comment by uranusjr** [comment]
What’s the next step here? Do we need to keep this issue open?

**Comment by RoyLee1224** [comment]
We've successfully finished all the original sub-issues and completed the content translations.

Regarding the topic of adding any other language support, that seems like a new requirement outside the original scope.I think we can then close this issue.
